src/main/clojure/aerospike_clj/protocols.clj

```diff
@@ -154,9 +164,9 @@
 (defprotocol ClientEvents
   "Continuation functions that are registered when an async DB operation is called.
   The value returned from those function will be the value of the returned future from the async operation."
-  (on-success [this op-name op-result index op-start-time]
+  (on-success [this op-name op-result index op-start-time ctx]
```

Comment:
I think that if we're going in the direction of a `ctx` then we might as well wrap everything in a map and pass it, so the protocol would be:
```clojure
(defprotocol ClientEvents
  (on-success [this ctx])
  (on-failure [this ctx]))
```

Where we guarantee a minimal schema of `ctx` in each case.
```